src/Functions/DateTimeTransforms.h

```diff
@@ -407,6 +407,23 @@ struct ToHourImpl
     using FactorTransform = ToDateImpl;
 };
 
+struct timezoneOffsetImpl
+{
+    static constexpr auto name = "timezoneOffset";
+
+    static inline time_t execute(UInt32 t, const DateLUTImpl & time_zone)
+    {
+        return time_zone.timezoneOffset(t);
+    }
+
+    static inline time_t execute(UInt16, const DateLUTImpl &)
+    {
+        return dateIsNotSupported(name);
+    }
+/////need to do
+    using FactorTransform = ToDateImpl;
```

Comment:
Unfortunately, we even can't use `ToDateImpl` because this function is not **positively** monotonic (as assumed for other transforms). We should update `getMonotonicityForRange`
https://github.com/ClickHouse/ClickHouse/blob/f897f7c93ff7be363dafe3de5a593dcb457251a1/src/Functions/FunctionDateOrDateTimeToSomething.h#L123
or just use `ToTimeImpl` for now.

Reply:
I use ToTimeImpl now.